sgrep/lib/normalize_generic.ml

```diff
@@ -71,11 +71,21 @@ let rec eval x =
   | L x -> Some x
   | Id (_, { id_const_literal = {contents = Some x}; _}) -> Some x
 
-  | Call(IdSpecial((ArithOp(Plus) | Concat), _), [Arg e1; Arg e2]) ->
-    (match eval e1, eval e2 with
-    | Some (String (s1, t1)), Some (String (s2, _t2)) ->
-          Some (String (s1 ^ s2, t1))
-    | _ -> None
+  | Call(IdSpecial((ArithOp(Plus) | Concat), _), args) ->
+    let literals = args |> Common.map_filter (fun (arg) ->
```

Comment:
This is wrong.
The code should actually make sure all the arguments are an Arg that evaluates to a Some String.
If any of the argument is not a string, then the whole thing should fail, this is why
there is the | _ -> None case in my code, because if you look at the caller, if it's a None then the match
will call fail ()

Reply:
For example $F("...") should not match foo("foo" + bar) (assuming bar is not a constant).
but with your code it will.